core/src/test/java/org/apache/iceberg/TestTransaction.java

```diff
@@ -771,4 +773,88 @@ public void testSimpleTransactionNotDeletingMetadataOnUnknownSate() throws IOExc
     Assert.assertTrue("Manifest file should exist", new File(manifests.get(0).path()).exists());
     Assert.assertEquals("Should have 2 files in metadata", 2, countAllMetadataFiles(tableDir));
   }
+
+  @Test
+  public void testValidationPasses() {
+    Assert.assertEquals("Table should be on version 0", 0, (int) version());
+
+    Transaction txn = table.newTransaction();
+    AppendFiles appendFiles = txn.newAppend().appendFile(FILE_A);
+    appendFiles.validateCurrentTable(currentTable -> true, "Custom validation failed.");
+    appendFiles.commit();
+    txn.commitTransaction();
+
+    Assert.assertEquals("Table should be on version 1 after commit", 1, (int) version());
+  }
+
+  @Test
+  public void testValidationFails() {
+    Assert.assertEquals("Table should be on version 0", 0, (int) version());
+
+    Transaction txn = table.newTransaction();
+    AssertHelpers.assertThrows(
```

Comment:
usages of this class are deprecated now. Please use `Assertions.assertThatThrownBy(..).isInstanceOf(..).hasMessage(..)`

Reply:
Fixed.